### PR TITLE
fix(stable): improve the way the kernel calculates how many pages to allocate

### PR DESCRIPTION
**Pull request by zshipko**
Fixes a bug in the kernel memory allocator where the space used by the `MemoryRoot` wasn't being considered

### CODE REVIEW

**Review by wikiwong — APPROVED** [review]

File: rust/src/lib.rs [hunk 1]
```diff
@@ -319,4 +320,30 @@ mod tests {
...
+            None,
+            hello_world,
+        );
+
+        let context = Context::new();
+        let mut plugin = Plugin::new(&context, REFLECT_WASM, [f], true).unwrap();
+
+        for i in 1..65540 {
+            let input = "a".repeat(i);
+            println!("i: {i}");
```
Comment (bhelx) [inline rust/src/lib.rs:340]:
```suggestion
```

File: rust/src/lib.rs [hunk 2]
```diff
@@ -319,4 +320,30 @@ mod tests {
...
+            let output = plugin.call("reflect", input.clone());
+            if let Err(e) = &output {
+                println!("output: #{}", e);
+            }
```
Comment (bhelx) [inline rust/src/lib.rs:344]:
```suggestion
```

File: kernel/src/lib.rs [hunk 3]
```diff
@@ -121,7 +121,7 @@ pub struct MemoryBlock {
 pub fn num_pages(nbytes: u64) -> usize {
     let npages = nbytes / PAGE_SIZE as u64;
     let remainder = nbytes % PAGE_SIZE as u64;
-    if remainder != 0 || nbytes < PAGE_SIZE as u64 {
```
Comment (wikiwong) [inline kernel/src/lib.rs]: We no longer need to check if `nbytes < PAGE_SIZE` ?

Comment (zshipko) [reply kernel/src/lib.rs]: Yeah, @chrisdickinson pointed out in another comment that it's actually alright: https://github.com/extism/extism/pull/472#discussion_r1330777361

**Pull request merged**